### DISCUSSION TITLE
(De)Serialize files without a worktree

### DISCUSSION THREAD

**Discussion by thataboy** [zed-industries/zed#17249] [Feature Requests]
I like to open random files that don't belong in any worktree, and when I relaunch zed, I want them to reopen as they were. This include files like notes, copypastas, logs, files from other projects that I use for references, etc. 

With sublime, I can open any files, any folders, leave any buffers unsaved, close sublime, and when I relaunch, they all come back -- windows, files, buffers, saved or unsaved, in a project or not. It never asks me to save buffers unless I'm closing them specifically. I'm messy and I like working this way.

Currently, zed does not (de)serialize files without a worktree, AFAICT (only files in projects or unsaved buffers). Like if you open the keymap, it won't reopen on launch. I thought I could do something with items.rs: deserialize and switched out project.find_worktree with project.find_or_create_worktree. After slugging it out with the rust compiler, I got it to compile and run but got a bunch of errors from repository.rs and ProjectLspAdapterDelegate.

Am I just banging my head against the wall? zed is just not made to work this way?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
